Sources/MongoKitten/AggregateStage.swift

```diff
@@ -1,227 +1,277 @@
+        self.stage = ["$project": projection.document]
+        self.minimalVersionRequired = projection.minimalVersion
     }
     
-    public static func project(_ fields: String...) -> AggregateBuilderStage {
+    public init(_ fields: String...) {
         var document = Document()
         for field in fields {
             document[field] = Projection.ProjectionExpression.included.primitive
         }
         
-        return AggregateBuilderStage(document: [
-            "$project": document
-        ])
+        self.stage = ["$project": document]
     }
+}
+
+public struct Sort: AggregateBuilderStage {
+    public internal(set) var stage: Document
+    public internal(set) var minimalVersionRequired: WireVersion? = nil
     
-    public static func count(to field: String) -> AggregateBuilderStage {
-        return AggregateBuilderStage(document: [
-            "$count": field
-        ])
+    public init(document: Document) {
+        self.stage = document
     }
+}
+
+public struct Count: AggregateBuilderStage {
+    public internal(set) var stage: Document
+    public internal(set) var minimalVersionRequired: WireVersion? = .mongo3_4
     
-    public static func skip(_ n: Int) -> AggregateBuilderStage {
-        assert(n > -1)
-        
-        return AggregateBuilderStage(document: [
-            "$skip": n
-        ])
+    public init(to field: String) {
+        self.stage = ["$count": field]
     }
+}
+
+public struct Skip: AggregateBuilderStage {
+    public internal(set) var stage: Document
+    public internal(set) var minimalVersionRequired: WireVersion? = nil
     
-	/// The `limit` aggregation limits the number of resulting documents to the given number
-	///
-	/// # MongoDB-Documentation:
-	/// [Link to the MongoDB-Documentation](https://docs.mongodb.com/manual/reference/operator/aggregation/limit/)
-	///
-	/// # Example:
-	/// ```
-	/// let pipeline = myCollection.aggregate([
-	///     .match("myCondition" == true),
-	///     .limit(5)
-	/// ])
-	///
-	/// pipeline.execute().whenComplete { result in
-	///    ...
-	/// }
-	/// ```
-	///
-	/// - Parameter n: the maximum number of documents
-	/// - Returns: an `AggregateBuilderStage`
-    public static func limit(_ n: Int) -> AggregateBuilderStage {
-        assert(n > 0)
-        
-        return AggregateBuilderStage(document: [
-            "$limit": n
-        ])
+    public init(_ n: Int) {
+        self.stage = ["$skip": n]
     }
+}
+
+public struct Out: AggregateBuilderStage {
+    public internal(set) var stage: Document
+    public internal(set) var minimalVersionRequired: WireVersion? = nil
     
-    public static func sample(_ n: Int) -> AggregateBuilderStage {
+    public init(toCollection collectionName: String, in database: String? = nil) {
+        if let db = database {
+            self.stage = ["$out": ["db": db, "coll": collectionName ]]
+            self.minimalVersionRequired = .mongo4_4
+        }
+        self.stage = ["$out": collectionName]
```

Comment:
This stage will overwrite the stage defined in the if-statement.

Reply:
this is actually the intended behavior as far as I understand.

As described in the syntax section of the docs, $out either takes just the collection name and assumes we're talking about the database we are currently using, or, if we specify a database, follows a different structure.

https://docs.mongodb.com/manual/reference/operator/aggregation/out/#syntax

Reply:
Correct, but the first `if let` statement will be completely invalidated because this code is not in an `else` block. It's like the following code:

```swift
var i: Int

if someCondition() {
  i = 2
}

i = 3
```

Even if the condition is `true`, the value of `i` is always 3.